linen_examples/imagenet/imagenet_lib.py

```diff
@@ -210,7 +212,29 @@ def save_checkpoint(state, model_dir):
 def sync_batch_stats(state):
   """Sync the batch statistics across replicas."""
   avg = jax.pmap(lambda x: lax.pmean(x, 'x'), 'x')
-  return state.replace(batch_stats=avg(state.batch_stats))
+
+  new_model_state = state.model_state.copy(
+      avg(state.model_state['batch_stats']))
```

Comment:
Shouldn't it be
```
  new_model_state = state.model_state.copy(
      dict(batch_stats=avg(state.model_state['batch_stats'])))
```

I wonder if this function is covered by the unittest...

Reply:
Yes I screwed up after a rebase. Nice catch :)